### ISSUE TITLE
Batch dispatch seems to not be working

### ISSUE THREAD

**Issue by thiagokokada** [labi-le/hyprland-ipc-client#7]
**Describe the bug**

I am trying to dispatch multiple commands via IPC, but it seems to not be working correctly.

**To Reproduce**

0. I found the easiest way to reproduce this using `exec` dispatch, but this issue happens every time you try to dispatch multiple commands
1. Apply this patch
```patch
diff --git a/ipc_test.go b/ipc_test.go
index 4691a78..818c3fb 100755
--- a/ipc_test.go
+++ b/ipc_test.go
@@ -182,7 +182,8 @@ func Test_ipc_CursorPos(t *testing.T) {
 
 func TestIpc_Dispatch(t *testing.T) {
 	q := NewByteQueue()
-	q.Add([]byte("exec"))
+	q.Add([]byte("exec kitty"))
+	q.Add([]byte("exec kitty"))
 	_, err := ipctest.Dispatch(q)
 	if err != nil {
 		t.Error(err)
```
2. `go test -run TestIpc_Dispatch -v`
3. `kitty` will start only once, and that instance shows `Failed to launch child: exec`. It seems to imply that `exec kitty` twice made starting `kitty exec kitty` instead.

**Expected behavior**
Two `kitty` instances would be open instead.

**Keep in mind** that this is an example. I am not that interested in the `exec` dispatch, I just want batch dispatch to work.

**Screenshots**
![2024-07-21-132115_hyprshot](https://github.com/user-attachments/assets/64f0aed6-2197-4219-bd84-c416eebe1a38)

**Desktop (please complete the following information):**
 - OS: NixOS
 - Hyprland: 0.42.2

**Additional context**
N/A.

**Comment by thiagokokada** [comment]
I found the issue. If you print `q.Glue()` results with the patch above, you will get:
```
[[BATCH]] dispatch exec kitty exec kitty
```

This is wrong, it should be something like:

```
[[BATCH]] dispatch exec kitty; dispatch exec kitty
```

So you need to do something like this:

```patch
diff --git a/ipc_test.go b/ipc_test.go
index 4691a78..4f03326 100755
--- a/ipc_test.go
+++ b/ipc_test.go
@@ -182,7 +182,8 @@ func Test_ipc_CursorPos(t *testing.T) {
 
 func TestIpc_Dispatch(t *testing.T) {
 	q := NewByteQueue()
-	q.Add([]byte("exec"))
+	q.Add([]byte("exec kitty;"))
+	q.Add([]byte("dispatch exec kitty"))
 	_, err := ipctest.Dispatch(q)
 	if err != nil {
 		t.Error(err)

```

It works, but this is of course really counter-intuitive.

**Comment by thiagokokada** [comment]
Another issue is that there is no validation about the maximum number of commands in batch, e.g.:

```patch
diff --git a/ipc_test.go b/ipc_test.go
index 4691a78..23d3ed1 100755
--- a/ipc_test.go
+++ b/ipc_test.go
@@ -182,7 +182,10 @@ func Test_ipc_CursorPos(t *testing.T) {
 
 func TestIpc_Dispatch(t *testing.T) {
 	q := NewByteQueue()
-	q.Add([]byte("exec"))
+	q.Add([]byte("exec kitty;"))
+	for i := 0; i < 32; i++ {
+		q.Add([]byte("dispatch exec kitty;"))
+	}
 	_, err := ipctest.Dispatch(q)
 	if err != nil {
 		t.Error(err)

 		t.Error(err)
```

Running this above will result in only one Kitty instance being opened. If you set `i < 31` it is fine (yes, I am doing some weird things with the IPC that needs to trigger multiple commands in batch mode). BTW, the following works in `hyprctl`:

```console
hyprctl --batch "dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;dispatch exec kitty;"
```

I imagine the difference is that `hyprctl` is splitting the batch in multiple smaller batches, but I am not sure. Ideally this project should do the same, but if not at least an error or panic should be generated if the user pass the maximum number of commands.

**Comment by labi-le** [comment]
Thank you so much for your input, I looked at the code, looks good, hope nothing is broken
